From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Byron Marohn <combustible@live.com>
Date: Thu, 26 Oct 2017 22:09:28 -0700
Subject: [PATCH] Monumenta - Block behavior changes

- Prevent ice from breaking/melting into water
- Prevent concrete powder from solidifying
- Coral is always waterlogged

Signed-off-by: Byron Marohn <combustible@live.com>

diff --git a/src/main/java/net/minecraft/world/level/block/IceBlock.java b/src/main/java/net/minecraft/world/level/block/IceBlock.java
index 5ecf02ce83b7496c977adfeb203b8eadb05f9da5..6fdbc59a8035330bbfa5d83c46797a012fc4ed47 100644
--- a/src/main/java/net/minecraft/world/level/block/IceBlock.java
+++ b/src/main/java/net/minecraft/world/level/block/IceBlock.java
@@ -31,16 +31,7 @@ public class IceBlock extends HalfTransparentBlock {
     public void afterDestroy(Level world, BlockPos pos, ItemStack tool) {
         // Paper end
         if (EnchantmentHelper.getItemEnchantmentLevel(Enchantments.SILK_TOUCH, tool) == 0) {
-            if (world.dimensionType().ultraWarm()) {
-                world.removeBlock(pos, false);
-                return;
-            }
-
-            Material material = world.getBlockState(pos.below()).getMaterial();
-
-            if (material.blocksMotion() || material.isLiquid()) {
-                world.setBlockAndUpdate(pos, Blocks.WATER.defaultBlockState());
-            }
+            world.removeBlock(pos, false);
         }
 
     }
@@ -54,17 +45,7 @@ public class IceBlock extends HalfTransparentBlock {
     }
 
     protected void melt(BlockState state, Level world, BlockPos pos) {
-        // CraftBukkit start
-        if (org.bukkit.craftbukkit.event.CraftEventFactory.callBlockFadeEvent(world, pos, world.dimensionType().ultraWarm() ? Blocks.AIR.defaultBlockState() : Blocks.WATER.defaultBlockState()).isCancelled()) {
-            return;
-        }
-        // CraftBukkit end
-        if (world.dimensionType().ultraWarm()) {
-            world.removeBlock(pos, false);
-        } else {
-            world.setBlockAndUpdate(pos, Blocks.WATER.defaultBlockState());
-            world.neighborChanged(pos, Blocks.WATER, pos);
-        }
+        return;
     }
 
     @Override
